### DISCUSSION TITLE
Detecting bootc hosts

### DISCUSSION THREAD

**Discussion by cgwalters** [bootc-dev/bootc#242] [Q&A] [answered]
Proxying this from a chat request: What's the best practice to detect a bootc-enabled host system?

Today we have `/run/ostree-booted` etc. - but in the future I think we want to get away from that.

One thing I was thinking about here is leveraging `/usr/lib/os-release` - we could add a field like `UPDATE_MECHANISM=bootc`.  And I'd advocate doing e.g. `UPDATE_MECHANISM=dnf` for package-based systems.

For now the spec suggests distro-specific fields, so for e.g. Fedora derivatives we could do `FEDORA_UPDATE_MECHANISM=bootc` and `FEDORA_UPDATE_MECHANISM=dnf`.

That said...it may be simplest to add a stub `bootc.service` unit here in preparation for https://github.com/containers/bootc/issues/2

Yet another approach would be having `bootc status` never fail, but output a clear state value.  Of these, I like this the most as far as being immediately actionable.

**Comment by beav** [comment]
I'd be +1 to the last option as well.

What would the `bootc status --json` output look like for a system with bootc installed but not in use? If there is a boolean or string that we can guarantee will exist, we can use that to detect if we should process the other fields or stop.

**Reply by cgwalters** [reply, marked as answer]
OK I've put up https://github.com/containers/bootc/pull/244
